support/client/lib/vwf/utility/vertexSimplify/simplify.js

```diff
@@ -0,0 +1,188 @@
+/*
+ (c) 2013, Vladimir Agafonkin
+ Simplify.js, a high-performance JS polyline simplification library
+ mourner.github.io/simplify-js
+*/
+define ( function() { 
+    'use strict';
+    // Modified from original Simplify.js npm module to correct script errors thrown on define
+
+    var Simplify = {
+        // to suit your point format, run search/replace for '.x' and '.y';
+        // for 3D version, see 3d branch (configurability would draw significant performance overhead)
+
+        // square distance between 2 points
+        getSqDist: function(p1, p2) {
+
+            var dx = p1.x - p2.x,
+                dy = p1.y - p2.y;
+
+            return dx * dx + dy * dy;
+        },
+
+        // square distance from a point to a segment
+        getSqSegDist: function(p, p1, p2) {
+
+            var x = p1.x,
+                y = p1.y,
+                dx = p2.x - x,
+                dy = p2.y - y;
+
+            if (dx !== 0 || dy !== 0) {
+
+                var t = ((p.x - x) * dx + (p.y - y) * dy) / (dx * dx + dy * dy);
+
+                if (t > 1) {
+                    x = p2.x;
+                    y = p2.y;
+
+                } else if (t > 0) {
+                    x += dx * t;
+                    y += dy * t;
+                }
+            }
+
+            dx = p.x - x;
+            dy = p.y - y;
+
+            return dx * dx + dy * dy;
+        },
+        // rest of the code doesn't care about point format
+
+        // basic distance-based simplification
+        simplifyRadialDist: function(points, sqTolerance) {
+
+            var prevPoint = points[0],
+                newPoints = [prevPoint],
+                point;
+
+            for (var i = 1, len = points.length; i < len; i++) {
+                point = points[i];
+
+                if (this.getSqDist(point, prevPoint) > sqTolerance) {
+                    newPoints.push(point);
+                    prevPoint = point;
+                }
+            }
+
+            if (prevPoint !== point) newPoints.push(point);
+
+            return newPoints;
+        },
+
+        // simplification using optimized Douglas-Peucker algorithm with recursion elimination
+        simplifyDouglasPeucker: function(points, sqTolerance) {
```

Comment:
I bet Douglas Peucker got made fun of quite a bit as a child....

Reply:
Probably had a hard time getting dates.  :-)